<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/home/marius/Downloads/rgbd_dataset_freiburg2_xyz-2hz-with-pointclouds.bag"/>
  
  <arg name="voxel_size" default="0.01"/>

  <node name="player" pkg="rosbag" type="play" output="screen" args=" -r 1.0 -s 0.6 --clock $(arg bag_file)" if="$(arg play_bag)"/>

  <node name="voxblox_node" pkg="voxblox_ros" type="segmentation_server" output="screen" args="--alsologtostderr" clear_params="true" >
    <remap from="pointcloud" to="/camera/rgb/points"/>
<!--    <remap from="pointcloud" to="/manipulation_rgbd_cam/depth/color/points"/>-->
    
    <remap from="depth_camera_info" to="/camera/depth/camera_info"/>
<!--    <remap from="depth_camera_info" to="/manipulation_rgbd_cam/depth/camera_info"/>-->

    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="true" />
    <remap from="transform" to="/kinect/vrpn_client/estimated_transform" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <param name="verbose" value="true" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="max_ray_length_m" value="2.0" />
    <param name="method" value="fast" />
    <param name="start_voxel_subsampling_factor" value="1.5" />
    <param name="max_consecutive_ray_collisions" value="5" />

    <param name="seg_canny_low_tresh" value="50" />
    <param name="seg_canny_high_tresh" value="150" />
    <param name="seg_canny_kernel_size" value="3" />
    <param name="seg_min_concavity" value="0.97" />
    <param name="seg_max_dist_step" value="0.005" />
    
    <param name="seg_voxel_prop_radius" value="2" />
    
    <param name="min_segment_pixel_size" value="20" />
    <param name="min_merge_confidence" value="4" />
    <param name="min_segment_merge_overlap" value="0.2" />
    <param name="min_segment_overlap" value="0.3" />

    <rosparam file="$(find voxblox_ros)/cfg/rgbd_dataset.yaml"/>

    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon rgbd).ply" />
  </node>
</launch>

<!--launch-prefix="valgrind -v"-->